aiopg/connection.py

```diff
@@ -449,7 +449,11 @@ def echo(self):
 
     if PY_341:  # pragma: no branch
         def __del__(self):
-            if not self._conn.closed:
+            try:
+                _conn = self._conn
+            except AttributeError:
+                return
+            if _conn is not None and not _conn.closed:
```

Comment:
May be this check should be moved inside `self.close()`, so finalizing method will be idempotent?